src/app/Http/Controllers/ApiController.php

```diff
@@ -51,10 +52,26 @@ public function locations(Request $request) {
             $locations->closeTo($lat,$lng);
         }
 
-        $locations = $locations->paginate(env('LOCATION_PAGE_SIZE', 100))->appends(compact([
```

Comment:
What about just changing this one line? 

```
$locations = $locations->paginate(min(2000, $request->input('limit', env('LOCATION_PAGE_SIZE', 100))))->appends(compact([
```

It may be cleaner to make a variable for limit above as you did. It's like to keep the page size control in an environment variable still, though. We could default this page size to 50 or 100 if no limit is specified but request more in Welcome.vue--that might be a good idea.

Reply:
Great call on this. See the updated version and let me know what you think!